0004-post-tags.md

```diff
@@ -0,0 +1,299 @@
+- Feature Name: Post Tags
+- Start Date: 2023-11-04
+- RFC PR: [LemmyNet/rfcs#0004](https://github.com/LemmyNet/rfcs/pull/4)
+- Lemmy Issue: [LemmyNet/lemmy#317](https://github.com/LemmyNet/lemmy/issues/317)
+
+# Summary
+
+Ability to add descriptive tags to Posts similar to Reddit's Flair system primarily on a per-community basis.
+
+# Motivation
+
+Post Tags would allow for vastly improved content filtering (and to a lesser extent searching) both for Users and Admins/Moderators. 
+
+# Guide-level explanation
+
+Tags grant you more fine grained control over the content you see on Lemmy. Certain users do not want to see certain content and tags allow you to do exactly that.
+
+For example a lot of communities offer mixed content such as Memes, News and Discussions. Some people would like to only see News or Discussions. In such a case they could blacklist (filter out) the "Meme" tag, going forward they would not be presented with any Memes in that community.
+
+Another use for tags is better handling of how posts are displayed. For example some people would like the content of a post to stay hidden for all users. This could be the case in a community focused on release discussions of Books, Movies, TV Shows or communities with content some users might find offensive but which is not neccesarily NSFW (for example jokes about certain topics).
+
+Essentially the tags serve as an easier way to message what is *inside* the post. Combining multiple tags allows for even finer control. Made a post about news in poland? Add the "News" and "Poland" tags to it and people interested exclusively in polish news can find it easily or people not interested can avoid it just as easily.
+
+# Reference-level explanation
+
+### Protocol:
+According to https://www.w3.org/TR/activitystreams-vocabulary/#dfn-tags a general tag object exists in the ActivityPub protocol. 
+Entirely unmoderated tags are not an option for lemmy as the moderation workload would be too much. Additionally users being able to type out tags themselves introduces splintering in the tag contents due to typos. A better solution is a curated list of tags users can attach to their posts. The list of tags can be maintained by both admins and moderators allowing for each community to tailor tags to their specific needs.
+Content for both sets of tags would be located in their respective root:
+
+- Instance Tags: `https://example.org/t/tag`
+- Community Tags: `https://example.org/c/instance/t/tag`
```

Comment:
The basic tag functionality is already complicated enough especially with federation in the mix. So I would get rid of the instance tags and move them to future possibilities instead. For the initial implementation its enough if each community can have its own tags.

Reply:
Agree that it makes sense to restrict to only community tags for now because I'm already thinking of many other things it could be extended with that would make it a behemoth

Reply:
I can think of 5 things which can be potentially tagged, some of these have issues already:

- Users: https://github.com/LemmyNet/lemmy/issues/1456
- Communities: https://github.com/LemmyNet/lemmy/issues/2675
- Posts: https://github.com/LemmyNet/lemmy/issues/317
- Instances
- Comments

Its pretty unclear which of these tags would be created (and applied) by either users, community mods, or instance admins.

Reply:
Moved Instance tags to the future outlook section and added some of the other options mentioned. (see ec854369f6c2306b16752e3ce3e650ea73983583)

Closing as resolved